api/daemon.go

```diff
@@ -129,42 +129,48 @@ func checkForUpdate() (bool, string, error) {
 }
 
 // applyUpdate downloads and applies an update.
-func applyUpdate(version string) (err error) {
+func applyUpdate(version string) error {
 	manifest, err := fetchManifest(version)
 	if err != nil {
-		return
+		return err
 	}
 
 	// Get the executable directory.
 	binDir, err := osext.ExecutableFolder()
 	if err != nil {
-		return
+		return err
 	}
 
-	// Initialize the updater object.
-	up, err := update.New().VerifySignatureWithPEM([]byte(developerKey))
-	if err != nil {
-		// should never happen
-		return
-	}
+	// Configure the update.
+	opts := new(update.Options)
+	opts.SetPublicKeyPEM([]byte(developerKey))
 
 	// Perform updates as indicated by the manifest.
 	for _, file := range manifest[1:] {
+		// set update path
+		opts.TargetPath = filepath.Join(binDir, file)
+
 		// fetch the signature
-		var sig []byte
-		sig, err = getHTTP(version, file+".sig")
+		opts.Signature, err = getHTTP(version, file+".sig")
```

Comment:
would it be easier to test if you created a helper function which did the actual verifying and updating (and had no http stuff in it)? Then the important stuff is at least covered.

A second question, do you switch keys every update? That's probably good practice, though I believe we're okay without it for the time being.